### PR TITLE
fix: Updated the Dropdown component's Usage tab to address a11y comments

### LINKED ISSUE #3526: Dropdown: Usage page

### PR DESCRIPTION
**Pull request by Kritvi-bhatia17**
Closes #3526 

This PR updates the Dropdown component Usage tab to incorporate comments from the referenced issue. Additionally, the content on the Usage tab has been aligned with the latest focus state discussion.

#### Changelog

**Changed**

- Content within several sections has been updated to address the comments (Ref: [Figma file](https://www.figma.com/design/xHzG8VCEAXXJnV0GAzcj44/Dropdown?node-id=5-64094&t=4aMoth3WRkE9kGmj-4)).
- Updated the images with latest focus state discussion.

### CODE REVIEW

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Only a couple small changes!

File: src/pages/components/dropdown/usage.mdx [hunk 1]
```diff
@@ -617,15 +620,19 @@ Use when you can select only one option from a list at a time.
...
 
-Use when you can select multiple options from a list or to filter information.
+Use a multiselect when the user needs to select multiple options from a list of
+predefined options. Multiselects are a good option when the user needs to filter
+or sort content on a page based on multiple criteria. A checkbox icon is
+included for each option in the list to enable multiselection.
 
 #### Making a selection
 
 - By default, the dropdown displays placeholder text in the field when closed.
```
Comment (mbgower) [inline src/pages/components/dropdown/usage.mdx:630]: This sorta reads like there is always placeholder text, which has already been listed above as optional. How about...
```suggestion
- By default, the dropdown displays any placeholder text in the field when closed.
```

**Review by laurenmrice — CHANGES_REQUESTED** [review]
@Kritvi-bhatia17 
- The AI images (default/fluid) on the Usage tab seem to still have the top border turned on under the focus state. They should be off.
- In the content section, for the part about the carbon tooltip on ellipses option text, we currently do not show it this way in code; instead, we show a browser-based tooltip. This should be a separate design issue that we need to come to a decision on to have more consistency across our components with how we treat ellipses on text. If a carbon tooltip is the desired design from a11y, we can look into that. Since we don't show this in code right now for dropdowns, I would remove it from these docs for now because we want to show information here that is currently 1:1 with code.

**Review by laurenmrice — APPROVED** [review]

**Comment by Kritvi-bhatia17** [comment]
Hi @mbgower, @tay1orjones, @ariellalgilmore!
Could you please review this PR when you have a moment? Thanks!

**Review by tay1orjones — APPROVED** [review]
LGTM!

**Comment by Kritvi-bhatia17** [comment]
Hi @mbgower, could you please review this PR? Thanks!

**Pull request merged**